verification_test.go

```diff
@@ -0,0 +1,322 @@
+package neutrino
+
+import (
+	"crypto/sha256"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/gcs"
+	"github.com/btcsuite/btcutil/gcs/builder"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	chainParams = &chaincfg.RegressionNetParams
+
+	dummySignature = []byte{34, 56, 78, 99}
+)
+
+// TestVerifyBlockFilter tests that a filter is correctly inspected for validity
+// against a downloaded block.
+func TestVerifyBlockFilter(t *testing.T) {
```

Comment:
Solid test! 

In the past we never implemented this tyupe of verification as we just had a hunch that all the basses were covered. We then created the `ComputePkScript` when we moved to support pkScript based spend+confirmation notifications, knowing that if we were correct, we could eventually use that here once we were confident in our assumptions.

Reply:
It turns out that bare P2PK isn't handled correctly by `ComputePkScript`. Maybe there are more of the "standard" transaction types that aren't fully supported yet. I added a TODO for now and skipped any non-witness inputs. That way we can at least get the witness verification part in.